asapdiscovery-ml/asapdiscovery/ml/cli.py

```diff
@@ -1392,151 +1462,165 @@ def build_and_train_schnet(
+        "cpd_regex": cpd_regex,
+        "cache_file": ds_cache,
+        "overwrite": overwrite_ds_cache,
+        "grouped": grouped,
+        "for_e3nn": False,
+    }
 
-            # Gather all missing values
-            missing_vals = [err["loc"][0] for err in exc.errors()]
+    ds_splitter_config = {
+        "cache": ds_split_config_cache,
+        "overwrite_cache": overwrite_ds_split_config_cache,
+        "split_type": ds_split_type,
+        "grouped": grouped,
+        "train_frac": train_frac,
+        "val_frac": val_frac,
+        "test_frac": test_frac,
+        "enforce_one": enforce_one,
+        "rand_seed": ds_rand_seed,
+    }
+    loss_config = {
+        "cache": loss_config_cache,
+        "overwrite_cache": overwrite_loss_config_cache,
+        "loss_type": loss_type,
+        "semiquant_fill": semiquant_fill,
+    }
 
-            raise ValueError(
-                "Tried to build Trainer but missing required values: ["
-                + ", ".join(missing_vals)
-                + "]"
-            )
+    # Parse loss_dict
+    if loss_dict:
+        loss_dict = json.loads(loss_dict.read_text())
+
+    # Filter out None Trainer kwargs
+    trainer_kwargs = {
+        "optimizer_config": optim_config,
+        "model_config": model_config,
+        "es_config": es_config,
+        "ds_config": ds_config,
+        "ds_splitter_config": ds_splitter_config,
+        "loss_config": loss_config,
+        "auto_init": auto_init,
+        "start_epoch": start_epoch,
+        "n_epochs": n_epochs,
+        "batch_size": batch_size,
+        "target_prop": target_prop,
+        "cont": cont,
+        "loss_dict": loss_dict,
+        "device": device,
+        "output_dir": output_dir,
+        "use_wandb": use_wandb,
+        "sweep": sweep,
+        "wandb_project": wandb_project,
+        "wandb_name": wandb_name,
+        "extra_config": Trainer.parse_extra_config(extra_config),
+    }
+    trainer_kwargs = {k: v for k, v in trainer_kwargs.items() if v is not None}
+
+    # If we got a config for the Trainer, load those args and merge with CLI args
+    if trainer_config_cache and trainer_config_cache.exists():
+        print("loading trainer args from cache", flush=True)
+        config_trainer_kwargs = json.loads(trainer_config_cache.read_text())
+
+        for config_name, config_val in config_trainer_kwargs.items():
+            # Arg wasn't passed at all, so got filtered out before
+            if config_name not in trainer_kwargs:
+                continue
+
+            if isinstance(config_val, dict):
+                config_val.update(
+                    {
+                        k: v
+                        for k, v in trainer_kwargs[config_name].items()
+                        if v is not None
+                    }
+                )
+            else:
+                config_trainer_kwargs[config_name] = trainer_kwargs[config_name]
+
+        trainer_kwargs = config_trainer_kwargs
```

Comment:
This has been repeated a lot, you can just make it a function that lives in this module so you only have to update one spot